pkg/operator/ceph/csi/ceph_config.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2022 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/daemon/ceph/client"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	configName = "csi-ceph-conf-override"
+
+	cephConfVal = `[global]
+auth_cluster_required = cephx
+auth_service_required = cephx
+auth_client_required = cephx
+
+# Workaround for http://tracker.ceph.com/issues/23446
+fuse_set_user_groups = false
+
+# ceph-fuse which uses libfuse2 by default has write buffer size of 2KiB
+# adding 'fuse_big_writes = true' option by default to override this limit
+# see https://github.com/ceph/ceph-csi/issues/1928
+fuse_big_writes = true
+`
+	// keyring should be adds to the same configmap as ceph.conf as we cannot
+	// create keyring file in the same directory as ceph.conf.
+	keyringKey = "keyring"
+	// empty keyring file content
+	keyringVal = ""
+)
+
+// CreateCsiConfigMap creates an ceph.conf for ceph-csi. If a config map already
+// exists, it will return it.
+func CreateCsiCephConfigMap(ctx context.Context, clientset kubernetes.Interface, ownerInfo *k8sutil.OwnerInfo, namespace string) error {
+	configMap := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configName,
+			Namespace: namespace,
+		},
+	}
+	configMap.Data = map[string]string{
+		client.DefaultConfigFile: cephConfVal,
+		keyringKey:               keyringVal,
+	}
+
+	err := ownerInfo.SetControllerReference(configMap)
```

Comment:
Yes, rook-operator deployment is the owner. yes once the rook operator is removed all CSI resources get deleted along with this configmap also.

Reply:
Yes but we don't want to throw user changes away...

Reply:
if we don't set owner ref this will be left when the Rook ceph cluster is teardowned. won't it be a cleanup problem?

Reply:
When the ceph cluster is teardowned the CM won't be deleted since it is not the owner, did you mean complete teardown operator included? I guess it depends, if the ns goes away it's fine, if the just cephcluster/operator goes away then the CM will remain...

Reply:
Yes, when the rook-operator deployment is deleted this should be deleted isn't it? or else the CM even with default configurations will be left in the namespace even if it's created by the rook operator.  I thought leaving stale resources is the problem for cleanup

Reply:
I'd argue it's probably better to leave the CM than trash the user's settings.

Reply:
I don't have any other option :+1: removed owner reference.